nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/util/writer/HadoopWriterContext.java

```diff
@@ -0,0 +1,202 @@
+        this.previousLatestFiles = previousLatestFiles;
+        this.writerFactory = writerFactory;
+        this.hdfsPrefix = hdfsPrefix;
+        this.logger = logger;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private ProcessSession session;
+        private FileStatusIterable fileStatusIterable;
+        private long minimumAge;
+        private long maximumAge;
+        private PathFilter pathFilter;
+        private FileStatusManager fileStatusManager;
+        private long previousLatestTimestamp;
+        private List<String> previousLatestFiles;
+        private Relationship successRelationship;
+        private RecordSetWriterFactory writerFactory;
+        private ComponentLog logger;
+        private String hdfsPrefix;
+
+        public Builder session(final ProcessSession session) {
+            this.session = session;
+            return this;
+        }
+
+        public Builder fileStatusIterable(final FileStatusIterable fileStatusIterable) {
+            this.fileStatusIterable = fileStatusIterable;
+            return this;
+        }
+
+        public Builder minimumAge(final long minimumAge) {
+            this.minimumAge = minimumAge;
+            return this;
+        }
+
+        public Builder maximumAge(final long maximumAge) {
+            this.maximumAge = maximumAge;
+            return this;
+        }
+
+        public Builder pathFilter(final PathFilter pathFilter) {
+            this.pathFilter = pathFilter;
+            return this;
+        }
+
+        public Builder fileStatusManager(final FileStatusManager fileStatusManager) {
+            this.fileStatusManager = fileStatusManager;
+            return this;
+        }
+
+        public Builder previousLatestTimestamp(final long previousLatestTimestamp) {
+            this.previousLatestTimestamp = previousLatestTimestamp;
+            return this;
+        }
+
+        public Builder previousLatestFiles(final List<String> previousLatestFiles) {
+            this.previousLatestFiles = previousLatestFiles;
+            return this;
+        }
+
+        public Builder successRelationship(final Relationship successRelationship) {
+            this.successRelationship = successRelationship;
+            return this;
+        }
+
+        public Builder writerFactory(final RecordSetWriterFactory writerFactory) {
+            this.writerFactory = writerFactory;
+            return this;
+        }
+
+        public Builder logger(final ComponentLog logger) {
+            this.logger = logger;
+            return this;
+        }
+
+        public Builder hdfsPrefix(final String hdfsPrefix) {
+            this.hdfsPrefix = hdfsPrefix;
+            return this;
+        }
+
+        public HadoopWriterContext build() {
+            return new HadoopWriterContext(session, successRelationship, fileStatusIterable, fileStatusManager, pathFilter, minimumAge, maximumAge,

```

Comment:
I would suggest to add some validation here which could check if all the mandatory attributes are set. Like for example: checking if `session` is not null.